aip/general/0236.md

```diff
@@ -0,0 +1,315 @@
+actually have the ability to alter traffic. Meanwhile, the output of the
+experimental version is logged so a customer can validate that the proposed
+changes are as intended. This gives the customer the ability to fully understand
+the effects of the new configuration against live traffic before replacing
+the effective configuration with it.
+
+### Non-goals
+
+#### Non-policy resources
+
+This rollout safety mechanism is for policies only.  
+
+A Policy is a customer-owned resource with a set of rules that govern the
+behavior of how other resources admit or deny a target entity  Generally, the
+outcome of a Policy can be evaluated to a specific set of outcomes. For example,
+a FirewallPolicy evaluates to either ALLOW or DENY for incoming traffic into
+a Virtual Network. This lends well to preview mode as a rollout safety mechanism
+as the proposed changes to a Policy can simply be logged instead of enforced.
+
+### Nested collection called experiments
+
+A new configuration of a Policy which is to be validated using preview mode is
+stored as a nested collection under that Policy resource.
+
+In this AIP, we will use the example of a hypothetical Policy resource called,
+`Policy`. The following is the pattern for the resource name:
+
+`projects/{project}/locations/{location}/policies/{policy}`
+
+The experimental versions of the resource used for preview mode are stored as a
+nested collection under `Policy`. The resource type will follow the convention
+"<resource_name> + Experiment”. The following is the pattern for the experiment:
+
+`/projects/{project}/locations/{location}/policies/{policy}/experiments/{experiment}`
+
+A proto used to represent an experiment **must** contain the following fields:
+
+  1. The required top-level fields for a resource, like `name` and `etag`.
+  2. The Policy that is being tested itself.
+  3. The `preview_metadata` field for preview.
+
+
+```proto
+message PolicyExperiment {
+
+  // The name of the PolicyExperiment.
+  string name = 1;
+ 
+  // The fields used for log evaluation and eventual promotion.
+  Policy policy = 2;
+
+  // Metadata for preview mode only.
+  PreviewMetadata preview_metadata = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+- A new proto **must** be used for experiments with the Policy proto inlined
+  inside.
+  - The inlined Policy field contains any changes to the live policy that are to
+    be tested using preview mode.
+- The name inside the embedded Policy message **must** be the name of the live
+  poliicy. 
+- When the user is ready to promote an experiment, the Policy message field
+  **must** be copied into the live resource and the experiment is deleted.
+  This can be done manually or via a custom method for commit described below.
+- A product **may** support multiple new configurations concurrently in preview
+  for a single live Policy. This is represented using two or more resources in
```

Comment:
Drop the sentence, "This is represented...in the nested collection." to be a sub-bullet in the list immediately below, a sibling of "The number of configurations...".